scripts/cluster.py

```diff
@@ -0,0 +1,58 @@
+# This script can be used to start Ray on an existing cluster.
+#
+# How to use it: Create a file "nodes.txt" that contains a list of the IP
+# addresses of the nodes in the cluster. Put the head node first. This node will
+# host the driver and the scheduler.
+
+import argparse
+import paramiko
+import IPython
+
+parser = argparse.ArgumentParser(description="Parse information about the cluster.")
+parser.add_argument("--nodes", default="nodes.txt", help="Test file with node IP addresses, one line per address.")
+parser.add_argument("--key-file", type=str, help="Path to the file that contains the private key.")
+parser.add_argument("--username", default="ubuntu", help="User name for logging in.")
+
+def connect(hostname, username, key):
+  c = paramiko.SSHClient()
+  c.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+  print "connecting to ", hostname
+  c.connect(hostname=hostname, username="ubuntu", pkey=key)
+  return c
+
+def install_ray_with(nodes, username, key):
+  connections = map(lambda hostname: connect(hostname, username, key), nodes)
+  for c in connections:
+    c.exec_command("sudo apt-get update; sudo apt-get -y install git; git clone https://github.com/amplab/ray; cd ray; bash setup.sh")
+
+def start_ray_with(nodes, username, key, worker_path):
+  head_conn = connect(nodes[0], username, key)
+  stdin, stdout, stderr = head_conn.exec_command("cd /home/ubuntu/ray/build; ./scheduler {}:10001".format(nodes[0]))
+  connections = map(lambda hostname: connect(hostname, username, key), nodes)
+  for (i, c) in enumerate(connections):
+    c.exec_command("cd ray/scripts; python start_workers.py --scheduler-address={}:10001 --node-ip={} --worker-path={}".format(nodes[0], nodes[i], worker_path))
+  print "cluster started; you can start the shell on the head node with:"
+  print "python shell.py --scheduler-address={}:10001 --objstore-address={}:20001 --worker-address={}:30001".format(nodes[0], nodes[0], nodes[0])
+  stdout.read()
+
+def stop_ray_with(nodes, username, key):
+  connections = map(lambda hostname: connect(hostname, username, key), nodes)
+  for c in connections:
+    c.exec_command("killall scheduler objstore python")
+
+if __name__ == "__main__":
+  args = parser.parse_args()
+  key = paramiko.RSAKey.from_private_key_file(args.key_file)
+  nodes = map(lambda s: str(s.strip()), open(args.nodes).readlines())
+
+  def install_ray(nodes):
```

Comment:
It will probably be important for the installation to give the user feedback about its progress. Right now it just prints 
```
In [1]: install_ray(nodes)
connecting to  52.51.27.104
connecting to  52.50.209.206

In [2]: 
```
and returns immediately, which makes it either look like it finished very quickly or it didn't do anything. At least we need to know when it finishes.